### PR TITLE
Added traits example for Scala

### PR DESCRIPTION
**Pull request by coastalchief**
added example on traits

### CODE REVIEW

File: scala.html.markdown [hunk 1]
```diff
@@ -6,6 +6,7 @@ contributors:
     - ["Dominic Bou-Samra", "http://dbousamra.github.com"]
     - ["Geoff Liu", "http://geoffliu.me"]
     - ["Ha-Duong Nguyen", "http://reference-error.org"]
+	- ["Christian Albrecht", "https://github.com/coastalchief"]
```
Comment (levibostian) [inline scala.html.markdown]: Can you fix the indentation here, please? This will break the site generator I think.

Comment (coastalchief) [inline scala.html.markdown]: done

File: scala.html.markdown [hunk 2]
```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
 // And many others. Case classes also get pattern matching for free, see below.
 
 
-// Traits coming soon!
+// Traits
+Traits are somewhat similar to Java interfaces. With traits you define
```
Comment (geoffliu) [inline scala.html.markdown]: This paragraph should be in comments.

File: scala.html.markdown [hunk 3]
```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
 // And many others. Case classes also get pattern matching for free, see below.
 
 
-// Traits coming soon!
+// Traits
+Traits are somewhat similar to Java interfaces. With traits you define  
+a type and method signatures. Other than Java interfaces, Scala allows
```
Comment (geoffliu) [inline scala.html.markdown]: "Other than" => "Unlike"

File: scala.html.markdown [hunk 4]
```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
...
+// Traits
+Traits are somewhat similar to Java interfaces. With traits you define  
+a type and method signatures. Other than Java interfaces, Scala allows  
+you to mix method signatures with method implementations.  
+Parameters for constructors are not allowed.  
+Traits can inherit from other traits or classes, but only if they have  
+no parameters.  
+```
+trait Bird {
+	def breed: String
```
Comment (geoffliu) [inline scala.html.markdown]: Rest of this page uses two spaces for indentation.

File: scala.html.markdown [hunk 5]
```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
...
+	def color: String
+	def saySomething: String = "tweet"
+	def canFly: Boolean
+}
+class Duck extends Bird{
```
Comment (geoffliu) [inline scala.html.markdown]: Space before the `{`

File: scala.html.markdown [hunk 6]
```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
...
+scala> d.canFly
+res1: Boolean = true 
+```
 
+// Traits could be used via Mixins (Keyword "with")  
+```
+trait Sound {
+	def saySomething: String = "tweet tweet"
+}
+trait Bird {
```
Comment (geoffliu) [inline scala.html.markdown]: No need to define `Bird` again. Also please do not redefine `Duck`, that's the whole point of mixins, that you don't need to redefine the classes every time. In the illustration below, just say `d = new Duck with Sound`

File: scala.html.markdown [hunk 7]
```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
...
+}
+class Duck extends Bird{
+	val breed = "bird"
+	val color = "green"
+	def canFly = true
+}
+```
+  
+```
+scala> var d = new Duck
```
Comment (geoffliu) [inline scala.html.markdown]: Use `val` whenever appropriate. Also, rest of this tutorial doesn't show output in the REPL style. We just do
    
    val d = new Duck
    d.breed  // "bird"

File: scala.html.markdown [hunk 8]
```diff
@@ -406,8 +407,63 @@ val otherGeorge = george.copy(phoneNumber = "9876")
...
+d: Duck = Duck@38c6f217  
+scala> d.breed
+res1: String = bird 
+scala> d.saySomething
+res0: String = tweet 
+scala> d.canFly
+res1: Boolean = true 
+```
 
+// Traits could be used via Mixins (Keyword "with")
```
Comment (geoffliu) [inline scala.html.markdown]: can be used as mixins.

**Comment by geoffliu** [comment]
@coastalchief Thanks for the edit, looks good in general, just a few minor stylistic issues. We can get this merged after you fix them.

**Comment by geoffliu** [comment]
@coastalchief What do you think? I would like to get this merged, since it's really useful stuff you wrote here.

**Comment by coastalchief** [comment]
thanks for the review, will do the fixes accordingly

**Comment by geoffliu** [comment]
@coastalchief Any progress on this?

**Comment by vendethiel** [comment]
@coastalchief ?

**Comment by geoffliu** [comment]
Closing for now. @coastalchief please reopen if you get around to it again.

**Pull request closed without merging**